spec/requests/users_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe '/users', type: :request do
+  describe 'UsersConroller' do
+    context 'GET index' do
+      before(:example) do
+        get '/users'
+      end
+
+      it 'Success response status for index action' do
+        expect(response).to be_successful
+      end
+
+      it 'Render correct templete for index action' do
+        expect(response).to render_template(:index)
+      end
+
+      it 'Correct body placeholder text for index action' do
+        expect(response.body).to include('Listing users')
+      end
+    end
+
+    context 'GET show' do
+      before(:example) do
+        get '/users/1'
+      end
+
+      it 'Success response status for show action' do
+        expect(response).to be_successful
+      end
+
+      it 'Render correct templete for show action' do
+        expect(response).to render_template(:show)
+      end
+
+      it 'Correct body placeholder text for show action' do
+        expect(response.body).to include('show user profile')
+      end
+    end
+  end
+end
```

Comment:
- Hi :clap:, Your test is failing because, in your UsersController, it can't find the user with the specified id. To fix this, you can update the show action to `def show; end` and remove the added `@user = User.find(params[:id])`. This should make your test pass as expected.


**NOTE:** This same issue applies to the PostsController.


![Screenshot from 2023-04-04 12-43-17](https://user-images.githubusercontent.com/59397159/229781022-006e559c-93bb-4826-a451-5942d4c2c57f.png)

Reply:
@zieeco , Thank you for the hint